src/content/ru/2021/pwa.md

```diff
@@ -0,0 +1,600 @@
+  sheets_gid="1466930372",
+  sql_file="lighthouse_pwa_score.sql"
+  )
+}}
+
+Вот несколько наблюдений:
+
+- Медианная оценка для «PWA-сайтов» составляет 83, по сравнению с 42 для «Всех сайтов».
+- В верхней части мы видим, что для «PWA-сайтов» не менее 10% получают максимальную (100) оценку в категории PWA. При просмотре «Всех сайтов» 75-й и 90-й процентили достигают значения не более 50.
+- Взглянув на нижнюю часть диаграммы, мы видим, что 90% «PWA-сайтов» имеют оценку PWA в Lighthouse не менее 50 по сравнению с 25 для «Всех сайтов».
+
+Опять же, разница между обеими группами является ожидаемой, поскольку «PWA-сайты», естественно, склонны проходить специфические для PWA требования чаще, чем «Все сайты». В любом случае средний балл 83 для PWA-сайтов предполагает, что значительная часть PWA-разработчиков придерживается лучших практик.
+
+## Библиотеки для работы с сервис-воркерами {service-worker-libraries}
+
+Сервис-воркеры могут использовать библиотеки для решения распространённых задач, применения функций и лучших практик (например, для реализации техник кэширования, push-уведомлений и т.д.). Наиболее распространенный способ делать это — использовать [importScripts()](https://developer.mozilla.org/ru/docs/Web/API/WorkerGlobalScope/importScripts), который является способом импорта JavaScript-библиотек в воркеры. В других случаях инструменты сборки также могут вставлять код библиотек прямо в сервис-воркеры во время сборки.
+
+Учтите, что не все библиотеки можно использовать в контекстах воркеров. У воркеров нет доступа к объекту [Window](https://developer.mozilla.org/ru/docs/Web/API/Window) и, следовательно, к объекту [Document](https://developer.mozilla.org/ru/docs/Web/API/Document), они имеют ограниченный доступ к браузерным API. По этой причине библиотеки для работы с сервис-воркерами специально разработаны для использования в этих контекстах.
+
+В этом разделе мы проанализируем популярность различных библиотек для работы с сервис-воркерами.
+
+### Популярные импортируемые скрипты {popular-import-scripts}
+
+Следующая диаграмма показывает процент использования различных библиотек, импортированных через `importScripts()`.
+
+{{ figure_markup(
+  image="pwa-libraries-and-scripts.png",
+  caption="Популярные PWA-библиотеки и скрипты.",
+  description="Гистограмма показывает популярные PWA-библиотеки и скрипты на десктопах и мобильных, с доминированием `workbox` на 15,43% десктопных сайтов и 16,58% мобильных, `recaptcha` — на 5,19% и 5,70% соответственно, `firebase` — на 2,45% и 2,63%, `OneSignalSDK` — на 2,53% и 2,51%, `sendpulse` — на 1,83% и 1,68%, `pushprofit` — на 1,42% и 1,65%, `quora` — на 1,28% и 1,07%, и, наконец, `sw_toolbox` — на 0,51% десктопных сайтов и 0,36% мобильных.", chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTHHnqNdpRUjoeTfsN9_irK57PvZn_Q2X842RLl-RL4ibWmZFvO-S1x35PjVE3-xUlHFS_Zurd22rOq/pubchart?oid=1758350320&format=interactive",
```

Comment:
We use long dash (`—`) instead of "is" or some missed words. If we have "15.43% on desktop and 2.45% on mobile", the next sentence says "5.19% on desktop, 5.70% on mobile". In English you just skip the words. In Russian we mark such skips with long dashes.